### PR TITLE
Feat: Changed user_interaction to class

### PR DESCRIPTION
**Pull request by SurprisedBird**

### CODE REVIEW

File: user_interaction.py [hunk 1]
```diff
@@ -8,65 +8,58 @@ class MessageScope(Enum):
...
 
-prepared_messages = {
-    MessageScope.GLOBAL: [],
-    MessageScope.ACTIVE: [],
-    MessageScope.PASSIVE: []
-}
+class UserInteraction():
+    def __init__(self):
 
+        self.prepared_messages = {
```
Comment (elias-gvin) [inline user_interaction.py]: Please, make prepared_messages protected. It SHOULD NOT be used outside UserInteraction class
self.prepared_messages -> self._prepared_messages

**Review by elias-gvin — CHANGES_REQUESTED** [review]

File: user_interaction.py [hunk 2]
```diff
@@ -8,65 +8,58 @@ class MessageScope(Enum):
     PASSIVE = auto()
 
 
-prepared_messages = {
-    MessageScope.GLOBAL: [],
-    MessageScope.ACTIVE: [],
-    MessageScope.PASSIVE: []
-}
+class UserInteraction():
+    def __init__(self):
```
Comment (elias-gvin) [inline user_interaction.py]: should contain reference to Context in future

**Review by elias-gvin — CHANGES_REQUESTED** [review]

File: user_interaction.py (hunk 1 above)
Comment (SurprisedBird) [reply user_interaction.py]: Done

File: user_interaction.py (hunk 2 above)
Comment (SurprisedBird) [reply user_interaction.py]: Done

Comment (asternotus) [reply user_interaction.py]: It should be add in the next general commit

File: user_interaction.py (hunk 1 above)
Comment (asternotus) [reply user_interaction.py]: Fixed all variables

File: utils.py [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
             msg.CommonMessages.ERROR_INVALID_TARGET)
-        target_number = user_interaction.read_number(message)
+        target_number = context.user_interaction.read_number(message)
 
     return target_number
 
 
-def save_message_for_player(game: 'Game', player: 'Citizen',
-                            message: str) -> None:
+def save_message_for_player(context: 'Context', game: 'Game',
```
Comment (elias-gvin) [inline utils.py]: You don't need game object here, since you have context

**Review by elias-gvin — CHANGES_REQUESTED** [review]

**Review by elias-gvin — APPROVED** [review]

**Pull request merged**